src/main/kotlin/org/jitsi/nlj/rtp/TransportCcEngine.kt

```diff
@@ -142,22 +156,69 @@ class TransportCcEngine(
         }
     }
 
+    /** Trim old sent packet details from the map.  Caller should be synchronized on [sentPacketsSyncRoot]. */
+    private fun cleanupSentPacketDetails(newSeq: Int): Boolean {
+        if (sentPacketDetails.isEmpty()) {
+            return true
+        }
+
+        val latestSeq = sentPacketDetails.lastKey()
+
+        /* If our sequence numbers have jumped by a quarter of the number space, reset the map. */
+        val seqJump = getSequenceNumberDelta(newSeq, latestSeq)
+        if (seqJump >= 0x4000 || seqJump <= -0x4000) {
+            sentPacketDetails.clear()
+            return true
+        }
+
+        val threshold = applySequenceNumberDelta(latestSeq, -MAX_OUTGOING_PACKETS_HISTORY)
+
+        if (isOlderSequenceNumberThan(newSeq, threshold)) {
+            return false
+        }
+
+        val it = sentPacketDetails.navigableKeySet().iterator()
+        while (it.hasNext()) {
+            val key = it.next()
+            if (isOlderSequenceNumberThan(key, threshold)) {
+                it.remove()
+            } else {
+                break
+            }
+        }
+
+        return true
+    }
+
     fun mediaPacketSent(tccSeqNum: Int, length: DataSize) {
         synchronized(sentPacketsSyncRoot) {
             val now = clock.instant()
-            sentPacketDetails.put(
-                tccSeqNum and 0xFFFF,
-                PacketDetail(length, now))
+            val seq = tccSeqNum and 0xFFFF
+            if (cleanupSentPacketDetails(seq)) {
```

Comment:
Why would we ever not want to put a newly-sent packet's information into `sentPacketDetails`?

Reply:
I copied this code from VP8FrameMap (and converted it to Kotlin); there, we had to worry about very out-of-order sequence numbers.  I guess that can't happen for TCC's, since they're generated locally, other than something very strange happening with threading; but I thought I'd still handle it.  I'll add a comment.